dora/core/client/fs/src/main/java/alluxio/client/file/FileSystemContext.java

```diff
@@ -967,11 +966,29 @@ public List<BlockWorkerInfo> getAllWorkers() throws IOException {
       List<BlockWorkerInfo> lostWorkers = masterClientResource.get().getLostWorkerInfoList()
           .stream().map(w -> new BlockWorkerInfo(w.getAddress(), w.getCapacityBytes(),
               w.getUsedBytes(), false)).collect(toList());
-      List<BlockWorkerInfo> allWorkers = new ArrayList<>();
-      allWorkers.addAll(liveWorkers);
-      allWorkers.addAll(lostWorkers);
-      return allWorkers;
+      // avoid duplicate elements in list
+      return combineAllWorkers(liveWorkers, lostWorkers);
+    }
+  }
+
+  private List<BlockWorkerInfo> combineAllWorkers(List<BlockWorkerInfo> liveWorkers,
+                                                  List<BlockWorkerInfo> lostWorkers) {
+    List<BlockWorkerInfo> allWorkers = new ArrayList<>();
+    for (BlockWorkerInfo liveWorker : liveWorkers) {
+      boolean isActive = true;
+      for (BlockWorkerInfo lostWorker : lostWorkers) {
```

Comment:
This search is more more expensive, two traverse, The complexity is m * n.
If set is used, the RB map is created, the complexity would be m * logn

Reply:
You're right, but `Set` depends on `equals` and `hashCode` methods. We don't want to change these two methods in `BlockWorkerInfo`.

Reply:
If the set is used, just check lostWorkerSet.contain

Reply:
The method `lostWorkerSet.contains` is based on the `equals` and `hashCode` methods os Set's element (`BlockWorkerInfo`). Therefore, if we don't change `equals` and `hashCode` methods, it doesn't work. But if put all `BlockWorkerInfo`'s fields into `equals` and `hashCode` methods, we can't distinguish the worker that is in 
 both `lostWorkerSet` and `liveWorkerSet`.

Reply:
I just updated the code to improve it with a `Map`.

Reply:
looks better.